pallets/clusters/src/lib.rs

```diff
@@ -29,40 +29,50 @@ pub struct ClusterMember {
 /// The **settings** of a **Role**
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct RoleSettings {
+	/// Name of the setting
 	pub name: Vec<u8>,
+	/// The data associated to the Role
 	pub data: Vec<u8>,
 }
 
 /// **Struct** of **Role** belonging to a **Cluster**.
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct Role {
+	/// Name of the role
 	pub name: Vec<u8>,
+	/// Settings of the Role
 	pub settings: RoleSettings,
-	pub members: Vec<ClusterMember>,
+	/// The list of Members associated to the role
+	pub members: Vec<Hash256>,
+	/// The optional list of Rules associated to the Role
 	pub rules: Option<Rule>,
 }
 
 /// **Struct** of **Rule** belonging to a **Role**.
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct Rule {
+	/// The name of the Rule
 	pub name: Vec<u8>,
 }
 
 /// **Struct** of a **Cluster**
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct Cluster<TAccountId> {
+	/// The owner of the Cluster
 	pub owner: TAccountId,
+	/// The name of the Cluster
 	pub name: Vec<u8>,
+	/// The list of Roles associated to the Cluster
 	pub roles: Vec<Hash256>,
```

Comment:
Hmm I'd say roles need to be embedded, not in a different structure.

Reply:
Right. Updated the cluster struct to embed roles as [`pub roles: Vec<Role<TAccountId>>`](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L60).